app/grandchallenge/emails/templates/vendored/mailgun_transactional_emails/action.html

```diff
@@ -1,49 +1 @@
" width="600" style="font-family: 'Helvetica Neue',Helvetica,Arial,sans-serif; box-sizing: border-box; font-size: 14px; vertical-align: top; display: block !important; max-width: 600px !important; clear: both !important; margin: 0 auto;" valign="top"> <div class="content" style="font-family: 'Helvetica Neue',Helvetica,Arial,sans-serif; box-sizing: border-box; font-size: 14px; max-width: 600px; display: block; margin: 0 auto; padding: 20px;"> <table class="main" width="100%" cellpadding="0" cellspacing="10" style="font-family: 'Helvetica Neue',Helvetica,Arial,sans-serif; box-sizing: border-box; font-size: 14px; border-radius: 3px; background-color: #fff; margin: 0; border: 1px solid #e9e9e9;" bgcolor="#fff"> <tr style="font-family: 'Helvetica Neue',Helvetica,Arial,sans-serif; box-sizing: border-box; font-size: 14px; margin: 0;"> <td class="content-wrap" style="font-family: 'Helvetica Neue',Helvetica,Arial,sans-serif; box-sizing: border-box; font-size: 14px; vertical-align: top; margin: 0; padding: 20px;" valign="top"> <table width="100%" cellpadding="0" cellspacing="0" style="font-family: 'Helvetica Neue',Helvetica,Arial,sans-serif; box-sizing: border-box; font-size: 14px; margin: 0;"> <p>Dear {{ username }}, </p> {{ content|md2email_html }} <tr style="font-family: 'Helvetica Neue',Helvetica,Arial,sans-serif; box-sizing: border-box; font-size: 14px; margin: 0;"> <td class="content-block" style="font-family: 'Helvetica Neue',Helvetica,Arial,sans-serif; box-sizing: border-box; font-size: 14px; vertical-align: top; margin: 0; padding: 0 0 20px;" valign="top"> Regards,<br> The {{ site.name }} Team </td> </tr> </table> </td> </tr> </table> <div class="footer" style="font-family: 'Helvetica Neue',Helvetica,Arial,sans-serif; box-sizing: border-box; font-size: 14px; width: 100%; clear: both; color: #999; margin: 0; padding: 20px;"> <table width="100%" style="font-family: 'Helvetica Neue',Helvetica,Arial,sans-serif; box-sizing: border-box; font-size: 14px; margin: 0;"> <tr style="font-family: 'Helvetica Neue',Helvetica,Arial,sans-serif; box-sizing: border-box; font-size: 14px; margin: 0;"> <td class="aligncenter content-block" style="font-family: 'Helvetica Neue',Helvetica,Arial,sans-serif; box-sizing: border-box; font-size: 12px; vertical-align: top; color: #999; text-align: center; margin: 0; padding: 0 0 20px;" align="center" valign="top"> This is an automated email from {{ site.domain.lower }}.<br> {% if unsubscribe_link %} If you no longer wish to receive {{ subscription_type|lower }} emails from us, <a href="{{ unsubscribe_link }}" style="font-family: 'Helvetica Neue',Helvetica,Arial,sans-serif; box-sizing: border-box; font-size: 12px; color: #999; text-decoration: underline; margin: 0;">unsubscribe here</a>. {% endif %} </td> </tr> </table> </div> </div> </td> <td style="font-family: 'Helvetica Neue',Helvetica,Arial,sans-serif; box-sizing: border-box; font-size: 14px; vertical-align: top; margin: 0;" valign="top"></td> </tr> </table> </body> </html>
```

Comment:
All the linebreaks and tabs have disappeared from this file, hard to read now. Was it intentional?

Reply:
Yes, because they otherwise end up as \n in the email.

Reply:
Okay, then I really do not understand what is going on. Why does `{{ content|md2email_html }}`, which can contain newlines, work then, but the template, which does contain newlines, does not?

Reply:
I also don't! Actually when trying things out again, even `{{ content|md2email_html }}` doesn't seem to work after all. Can't work out why it (seemed) to work before. I will try this out tomorrow again, with a fresh head. Sorry for the back and forth!

Reply:
I think there is a problem with your testing and there is no need to do any newline parsing. I cannot reproduce the problem. My process:
- Trigger the email which logs the raw email to the celery console
- Put the content into a `.eml` file
- Open that email with Apple Mail

Attached are the three emails that are produced when logging in as superuser:

`raw.html` created with 59e1f02:
[raw.eml.zip](https://github.com/comic/grand-challenge.org/files/14376766/raw.eml.zip)
<img width="2290" alt="Screenshot 2024-02-22 at 17 44 20" src="https://github.com/comic/grand-challenge.org/assets/12661555/cd60ba64-8ba4-4d02-bb08-227c51e3a7f6">

`raw-expanded.html` created with 883856f:
[raw-expanded.eml.zip](https://github.com/comic/grand-challenge.org/files/14376793/raw-expanded.eml.zip)
<img width="2269" alt="Screenshot 2024-02-22 at 17 46 26" src="https://github.com/comic/grand-challenge.org/assets/12661555/6cf7dd8a-0204-47d5-beb0-c4fb2ebb542e">

`raw-expanded-with-lots-of-newlines` created with 883856f but setting `markdown_message="First line\r\n\r\nSecond line\n\nWe noticed a new login to your account.\n\n\n\n\n\n\n\n\nIf this was you, you don't need to do anything. If not, please change your password and update your 2FA device."`:
[raw-expanded-with-lots-of-newlines.eml.zip](https://github.com/comic/grand-challenge.org/files/14376839/raw-expanded-with-lots-of-newlines.eml.zip)
<img width="2300" alt="Screenshot 2024-02-22 at 17 47 51" src="https://github.com/comic/grand-challenge.org/assets/12661555/0e162e7b-8a70-4f1e-9414-4f0c03834cd2">

Also did the same for the newsletter
[raw-newsletter.eml.zip](https://github.com/comic/grand-challenge.org/files/14376938/raw-newsletter.eml.zip)
<img width="2296" alt="Screenshot 2024-02-22 at 17 54 12" src="https://github.com/comic/grand-challenge.org/assets/12661555/9c9baa17-3aea-4759-9e2f-b08e5d57bce3">

Reply:
Ok, thanks for testing this as well! Looks great, I can also reproduce this now.

I had tested things in the shell (up until render_to_string with the action template). Saving the output of that to file and opening it in with Outlook looks fine for me as well, but I didn't test things that way, I instead copy pasted the rendered string into https://app.postdrop.io/ which apparently interprets the linebreaks differently.

Reply:
Okay, then yes, you needed to `print` the rendered string first before copying. Using the actual email document is more robust, you can also check the headers there too. Rather than copying from the console we could use the file backend, but this concatenates everything into a single file rather than individual messages https://docs.djangoproject.com/en/5.0/topics/email/#file-backend.